component/azstorage/block_blob.go

```diff
@@ -1123,25 +1129,42 @@ func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]*string, fi
 }
 
 // WriteFromBuffer : Upload from a buffer to a blob
-func (bb *BlockBlob) WriteFromBuffer(name string, metadata map[string]*string, data []byte) error {
-	log.Trace("BlockBlob::WriteFromBuffer : name %s", name)
-	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, name))
+func (bb *BlockBlob) WriteFromBuffer(options internal.WriteFromBufferOptions) error {
+	log.Trace("BlockBlob::WriteFromBuffer : name %s", options.Name)
+	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, options.Name))
 
-	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromBuffer", name)
+	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromBuffer", options.Name)
 
-	_, err := blobClient.UploadBuffer(context.Background(), data, &blockblob.UploadBufferOptions{
+	uploadOptions := &blockblob.UploadBufferOptions{
 		BlockSize:   bb.Config.blockSize,
 		Concurrency: bb.Config.maxConcurrency,
-		Metadata:    metadata,
+		Metadata:    options.Metadata,
 		AccessTier:  bb.Config.defaultTier,
 		HTTPHeaders: &blob.HTTPHeaders{
-			BlobContentType: to.Ptr(getContentType(name)),
+			BlobContentType: to.Ptr(getContentType(options.Name)),
 		},
 		CPKInfo: bb.blobCPKOpt,
-	})
+	}
+
+	if options.EtagNoneMatchConditions != "" {
+		uploadOptions.AccessConditions = &blob.AccessConditions{
+			ModifiedAccessConditions: &blob.ModifiedAccessConditions{
+				IfNoneMatch: to.Ptr(azcore.ETag(options.EtagNoneMatchConditions)),
```

Comment:
Use the constant from SDK
```suggestion
				IfNoneMatch: to.Ptr(azcore.ETagAny),
```